transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpSubjectSecurityFilter.java

```diff
@@ -112,6 +119,42 @@ public void doMessageReceived(NextFilter nextFilter, IoSession session, Object m
         if (! httpRequestMessageReceived(nextFilter, session, message)) return;
 
         HttpRequestMessage httpRequest = (HttpRequestMessage) message;
+        final boolean loggerIsEnabled = logger != null && logger.isTraceEnabled();
+
+        if (httpRequest.getHeader(HEADER_X_FORWARDED_FOR) == null) {
+            String remoteIpAddress = null;
+            SocketAddress socketAddress = session.getRemoteAddress();
+
+            if (socketAddress instanceof InetSocketAddress) {
+                InetAddress addr = ((InetSocketAddress) socketAddress).getAddress();
+                remoteIpAddress = addr.getHostAddress();
+
+                if (loggerIsEnabled) {
+                    logger.trace(format("HttpSubjectSecurityFilter: Remote IP Address: '%s'", remoteIpAddress));
+                }
+            }
+            else if (socketAddress instanceof ResourceAddress) {
+                ResourceAddress resourceAddress = (ResourceAddress) socketAddress;
+                ResourceAddress tcpResourceAddress = resourceAddress.findTransport("tcp");
+
+                if (tcpResourceAddress != null) {
+                    URI resource = tcpResourceAddress.getResource();
+
+                    // Resolve the hostname to the corresponding IP address.
```

Comment:
Looks like NIO resource address contains inetAddress.getHostAddress() and here we are resolving back to ip. I think we should stick to getting address from session hierarchy and get InetSocketAddress. Check with John.